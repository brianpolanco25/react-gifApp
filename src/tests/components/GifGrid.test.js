import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import { GifGrid } from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');

describe('Pruebas en <GifGrid />', () => {

    const category = 'category';

    test('debe cargar correctamente el componente', () => {

        useFetchGifs.mockReturnValue({
            data:[],
            loading: true
        });

        const wrapper = shallow( <GifGrid category={ category } /> );
        expect(wrapper).toMatchSnapshot();
    })
    
    test('debe de mostrar items cuando se cargan imagenes con useFetchGifs ', () => {

        const gifs = [{
            id: 'abbc',
            url: 'https://localhost/imagen.jpg',
            title: 'title'
        }]

        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        });

        const wrapper = shallow( <GifGrid category={ category } /> );

        expect(wrapper).toMatchSnapshot();
        expect( wrapper.find( 'p').exists() ).toBe(false);
        expect( wrapper.find('GifGridItem').length).toBe( gifs.length);
        
    })
    
    
})
