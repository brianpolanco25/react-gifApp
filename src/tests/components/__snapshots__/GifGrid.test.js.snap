// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGrid /> debe cargar correctamente el componente 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    category
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`Pruebas en <GifGrid /> debe de mostrar items cuando se cargan imagenes con useFetchGifs  1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    category
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="abbc"
      key="abbc"
      title="title"
      url="https://localhost/imagen.jpg"
    />
  </div>
</Fragment>
`;
